src/resource/PrivateResourceRoute.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { ReactNode, useEffect, useState } from 'react';
+import {
+  Route,
+  Redirect,
+  RouteProps,
+  RouteComponentProps,
+} from 'react-router-dom';
+import { AuthContextProps, useAuth } from '../auth/auth-context';
+import api from '../common/utils/api/api';
+
+type ComputedMatch = {
+  params: {
+    id?: string;
+  };
+};
+
+interface PrivateRouteProps extends RouteProps {
+  computedMatch?: ComputedMatch;
+  id: string;
+}
+
+export default function (routeProps: PrivateRouteProps): JSX.Element {
+  const { render, id, ...rest } = routeProps;
+  const resourceId = rest.computedMatch?.params.id;
+  const { authTokens, clearAuth }: Partial<AuthContextProps> = useAuth();
+  const [isLoading, setLoading] = useState<boolean>(true);
```

Comment:
We discussed about these loaders on slack. I wonder if it is a good idea to introduce yet another loader to the application? Or have those other loaders been reduced at this pr also? :D. Just wondering because we've had quite many loaders and it makes the whole page load look very clumsy.

Reply:
It actually replaces the one in the App.tsx